homeassistant/components/netatmo/webhook.py

```diff
@@ -0,0 +1,123 @@
+"""The Netatmo integration."""
+import logging
+
+from .const import (
+    ATTR_CAMERA_ID,
+    ATTR_EVENT_LIST,
+    ATTR_EVENT_TYPE,
+    ATTR_FACE_URL,
+    ATTR_HOME_NAME,
+    ATTR_ID,
+    ATTR_IS_KNOWN,
+    ATTR_MESSAGE,
+    ATTR_NAME,
+    ATTR_PERSONS,
+    ATTR_SNAPSHOT_URL,
+    ATTR_VIGNETTE_URL,
+    DATA_PERSONS,
+    DEFAULT_PERSON,
+    DOMAIN,
+    EVENT_ANIMAL,
+    EVENT_BUS_ANIMAL,
+    EVENT_BUS_CONNECTION,
+    EVENT_BUS_HUMAN,
+    EVENT_BUS_HUSH,
+    EVENT_BUS_MOVEMENT,
+    EVENT_BUS_OFF,
+    EVENT_BUS_ON,
+    EVENT_BUS_OTHER,
+    EVENT_BUS_OUTDOOR,
+    EVENT_BUS_PERSON,
+    EVENT_BUS_SMOKE,
+    EVENT_BUS_TAG_BIG_MOVE,
+    EVENT_BUS_TAG_OPEN,
+    EVENT_BUS_TAG_SMALL_MOVE,
+    EVENT_BUS_TAMPERED,
+    EVENT_BUS_VEHICLE,
+    EVENT_CONNECTION,
+    EVENT_HUMAN,
+    EVENT_HUSH,
+    EVENT_MOVEMENT,
+    EVENT_OFF,
+    EVENT_ON,
+    EVENT_OUTDOOR,
+    EVENT_PERSON,
+    EVENT_SMOKE,
+    EVENT_TAG_BIG_MOVE,
+    EVENT_TAG_OPEN,
+    EVENT_TAG_SMALL_MOVE,
+    EVENT_TAMPERED,
+    EVENT_VEHICLE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def handle_webhook(hass, webhook_id, request):
+    """Handle webhook callback."""
+    try:
+        data = await request.json()
+    except ValueError:
+        return None
+
+    _LOGGER.debug("Got webhook data: %s", data)
+
+    def evaluate_event(event_data):
```

Comment:
Please move this function definition out of the webhook handler. Just pass `hass` as another param.